src/authkit-callback-route.ts

```diff
@@ -5,56 +5,68 @@ import { WORKOS_CLIENT_ID } from './env-variables.js';
 import { encryptSession } from './session.js';
 import { cookieName, cookieOptions } from './cookie.js';
 
-export async function authkitCallbackRoute(request: NextRequest) {
-  const code = request.nextUrl.searchParams.get('code');
+interface HandleAuthOptions {
+  returnPathname?: string;
+}
+
+export function handleAuth(options: HandleAuthOptions = {}) {
+  const { returnPathname: returnPathnameOption = '/' } = options;
+
+  return async function GET(request: NextRequest) {
+    const code = request.nextUrl.searchParams.get('code');
+    const state = request.nextUrl.searchParams.get('state');
+    const returnPathname = state ? JSON.parse(atob(state)).returnPathname : null;
```

Comment:
What should happen if `state` doesn't contain valid JSON?

Right now it will throw, which is probably fine if the `state` was tampered with (maliciously or not), but wanted to point it out in case anyone had other preferences.

Reply:
Yeah I think it's fine for now considering it would only be an issue if tampered with.